### DISCUSSION TITLE
Support for SQLite Pocketbase plugins

### DISCUSSION THREAD

**Discussion by minfuel** [pocketbase/pocketbase#2274] [Polls]
## **Abstract**
A tab in pocketbase for extension/plugins. This unlocks many possibilities like being able to use existing SQLite plugins like this: [Some SQLite plugins](#some-sqlite-plugins) 

## **A suggestion of how it would look like:** 
![Untitled-2023-03-17-2310](https://user-images.githubusercontent.com/106936397/231367917-a6e1265e-ea60-40d7-a3f6-84662992882b.png)

## **Some SQLite plugins:**
1. [SQLite extension to query CSVs from other data sources, such the internet](https://github.com/asg017/sqlite-xsv) 
2. [SQLite extension for efficient vector search, based on Faiss!](https://github.com/asg017/sqlite-vss) 
3. [SQLite extension for making HTTP requests purely in SQL](https://github.com/asg017/sqlite-http)

**Comment by ganigeorgiev** [comment]
Even if you load an SQLite extension from the UI (_and this may not be possible with the Go transpiled port https://pkg.go.dev/modernc.org/sqlite that we use by default_) how users are expected to interact with those extensions? For example, how do you plan to use the CSV extension if there is no dedicated interface for it?

**Reply by minfuel** [reply]
@ganigeorgiev 
I see two options (_but open for changing them_)
- [ ] Compiling the extension into the SQLite library used by Pocketbase by modifying the go-sqlite3 package's source code to include the extension. **(_not sure if this is possible_)**
# **no dedicated interface**
Users could potentially use the SQL command interface within Pocketbase to interact with SQLite extensions that have been loaded into the database. For example, if a CSV extension has been loaded into the database, users could use SQL commands to query data from the CSV file. The specific syntax for interacting with the extension would depend on the extension and its functionality. The documentation for the extension should provide guidance on how to use it with SQL commands
